### PR TITLE
EnlsvgPathPlanner fix

### PR DESCRIPTION
**Pull request by itsarune**
### Description
Jon found a bug with EnlsvgPathPlanner that sometimes causes it to intersect an obstacle. It seems to come from EnlsvgAlgorithm mistaking the goalParent (the index of a point that it wants to travel to) for the destination

### Testing Done

- Added a test case that Jon showed me
- Existing tests (including enlsvgpathplanner tests pass)

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/test_util/path_planning_test_util.h [hunk 1]
```diff
@@ -41,4 +41,13 @@ namespace TestUtil
      */
     void checkPathDoesNotIntersectObstacle(std::vector<Point> path_points,
                                            std::vector<Polygon> obstacles);
+
+    /**
+     * Given a list of points, print it out to the console. ONLY mean to be used as a
+     * debugging tool.
```
Comment (akhilveeraghanta) [inline src/software/test_util/path_planning_test_util.h]: nit: you can remove the `ONLY mean to be used as a debugging tool`. Its in test_util so its implied

File: src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp [hunk 2]
```diff
@@ -690,6 +690,61 @@ TEST_F(TestEnlsvgPathPlanner,
...
+    // Make sure the start and end points match
+    EXPECT_EQ(start, path_two->getStartPoint());
+    EXPECT_EQ(dest, path_two->getEndPoint());
+
+    EXPECT_LE(4, path_points.size());
+
+    // Make sure path does not exceed a bounding box;
+    bounding_box = Rectangle(Point(5.1, 1.8), Point(-2.1, -0.1));
+    TestUtil::checkPathDoesNotExceedBoundingBox(path_points, bounding_box);
+    TestUtil::checkPathDoesNotIntersectObstacle(path_points, obstacles);
```
Comment (jonathanlew) [inline src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp]: this part isn't actually necessary, bad copy pasta on my part

**Review by akhilveeraghanta — COMMENTED** [review]
lgtm! nice regression test, just 1 nit

**Review by jonathanlew — COMMENTED** [review]
nice fix, just one comment

File: src/software/test_util/path_planning_test_util.h (hunk 1 above)
Comment (itsarune) [reply src/software/test_util/path_planning_test_util.h]: fixed

File: src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp (hunk 2 above)
Comment (itsarune) [reply src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp]: fixed

**Review by jonathanlew — COMMENTED** [review]
LGTM pending CI

**Review by akhilveeraghanta — APPROVED** [review]

**Review by jonathanlew — APPROVED** [review]

**Pull request merged**